### ISSUE TITLE
Properly document memory semantics for marshalling x-plat

### ISSUE THREAD

**Issue by AaronRobinsonMSFT** [dotnet/coreclr#19090]
One of the base assumptions on Windows was that `CoTaskMemAlloc()` and `CoTaskMemFree()` were the canonical memory management mechanism when performing an interop. These function do not exist on x-plat scenarios so we need some explicit documentation or example where this is demonstrated.

See https://github.com/dotnet/coreclr/issues/19082

**Comment by jkotas** [comment]
It should be documented under https://docs.microsoft.com/en-us/dotnet/framework/interop/interop-marshaling

**Comment by AaronRobinsonMSFT** [comment]
@jkotas Yep, but that is only the Windows _and_ I thought only for .NET Framework. This work is really creating an example or document that is related to CoreCLR and calls out these specific kind of differences.

Should I assume the referenced link also related to CoreCLR? I can add documentation there for the non-Windows case?

**Comment by jkotas** [comment]
Good point. I think the interop docs that are relevant across all platforms and runtimes should be moved from `.NET Framework Guide` to `.NET Guide`.

All public user facing documentation should be on  https://docs.microsoft.com/ . We should be updating it and augmentation as necessary for .NET Core and x-plat. For example, https://github.com/dotnet/docs/issues/6277 tracked inaccurate docs for env variables that have been fixed already.

The docs in CoreCLR repo should be about how to do CoreCLR development. They should not be duplicating the public user facing documentation.
